### ISSUE TITLE
Fabric 2.0.0-beta4: how to migrate old path-group to new group?

### ISSUE THREAD

**Issue by blobinabottle** [kangax/fabric.js#4136]
In Fabric 2, SVG (path-group) are now group. 
But how to handle old JSON loading SVG (path-group) previously generated by Fabric 1.x?
For the moment I get errors like this when loading previous JSON 
`at klass._enlivenObjects (editor.js:36651)
    at klass.loadFromJSON (editor.js:36551)`

Can we migrate old JSON to new ones? Or something like that? Is seems the whole structure of this kind of object has changed.

Thanks for your input!

**Comment by asturur** [comment]
Can you give me a pathGroup JSON so that i can base an example over it?
Thanks.

**Comment by blobinabottle** [comment]
Here is a JSON of on pathgroup. Hope it's helping? Thanks!
`{"background":"#FFF","hoverCursor":"move","objects":[{"angle":0,"backgroundColor":"","clipTo":null,"editable":true,"evented":true,"fill":"","fillRule":"nonzero","flipX":false,"flipY":false,"globalCompositeOperation":"source-over","height":48,"hoverCursor":"move","left":960,"lockScalingFlip":true,"lockUniScaling":true,"ludus":{"originalPaths":"{\"objects\":[{\"type\":\"path-group\",\"originX\":\"left\",\"originY\":\"top\",\"left\":0,\"top\":0,\"width\":48,\"height\":48,\"fill\":\"\",\"stroke\":null,\"strokeWidth\":1,\"strokeDashArray\":null,\"strokeLineCap\":\"butt\",\"strokeLineJoin\":\"miter\",\"strokeMiterLimit\":10,\"scaleX\":1,\"scaleY\":1,\"angle\":0,\"flipX\":false,\"flipY\":false,\"opacity\":1,\"shadow\":null,\"visible\":true,\"clipTo\":null,\"backgroundColor\":\"\",\"fillRule\":\"nonzero\",\"globalCompositeOperation\":\"source-over\",\"transformMatrix\":null,\"skewX\":0,\"skewY\":0,\"paths\":[{\"type\":\"path\",\"originX\":\"left\",\"originY\":\"top\",\"left\":6,\"top\":6,\"width\":36,\"height\":36,\"fill\":\"#2196F3\",\"stroke\":null,\"strokeWidth\":1,\"strokeDashArray\":null,\"strokeLineCap\":\"butt\",\"strokeLineJoin\":\"miter\",\"strokeMiterLimit\":10,\"scaleX\":1,\"scaleY\":1,\"angle\":0,\"flipX\":false,\"flipY\":false,\"opacity\":1,\"shadow\":null,\"visible\":true,\"clipTo\":null,\"backgroundColor\":\"\",\"fillRule\":\"nonzero\",\"globalCompositeOperation\":\"source-over\",\"transformMatrix\":null,\"skewX\":0,\"skewY\":0,\"pathOffset\":{\"x\":24,\"y\":24},\"path\":[[\"M\",42,37],[\"C\",42,39.800781,39.800781,42,37,42],[\"L\",11,42],[\"C\",8.199219,42,6,39.800781,6,37],[\"L\",6,11],[\"C\",6,8.199219,8.199219,6,11,6],[\"L\",37,6],[\"C\",39.800781,6,42,8.199219,42,11],[\"Z\"]]},{\"type\":\"path\",\"originX\":\"left\",\"originY\":\"top\",\"left\":21.5,\"top\":14,\"width\":5,\"height\":19,\"fill\":\"#FFFFFF\",\"stroke\":null,\"strokeWidth\":1,\"strokeDashArray\":null,\"strokeLineCap\":\"butt\",\"strokeLineJoin\":\"miter\",\"strokeMiterLimit\":10,\"scaleX\":1,\"scaleY\":1,\"angle\":0,\"flipX\":false,\"flipY\":false,\"opacity\":1,\"shadow\":null,\"visible\":true,\"clipTo\":null,\"backgroundColor\":\"\",\"fillRule\":\"nonzero\",\"globalCompositeOperation\":\"source-over\",\"transformMatrix\":null,\"skewX\":0,\"skewY\":0,\"pathOffset\":{\"x\":24,\"y\":23.5},\"path\":[[\"M\",26,33],[\"L\",22,33],[\"L\",22,22],[\"L\",26,22],[\"Z\"],[\"M\",26.5,16.5],[\"C\",26.5,15.101563,25.398438,14,24,14],[\"C\",22.601563,14,21.5,15.101563,21.5,16.5],[\"C\",21.5,17.898438,22.601563,19,24,19],[\"C\",25.398438,19,26.5,17.898438,26.5,16.5]]}]}]}","source":"icons8","tmpCanvas":"{\"objects\":[{\"type\":\"path-group\",\"originX\":\"left\",\"originY\":\"top\",\"left\":0,\"top\":0,\"width\":48,\"height\":48,\"fill\":\"\",\"stroke\":null,\"strokeWidth\":1,\"strokeDashArray\":null,\"strokeLineCap\":\"butt\",\"strokeLineJoin\":\"miter\",\"strokeMiterLimit\":10,\"scaleX\":1,\"scaleY\":1,\"angle\":0,\"flipX\":false,\"flipY\":false,\"opacity\":1,\"shadow\":null,\"visible\":true,\"clipTo\":null,\"backgroundColor\":\"\",\"fillRule\":\"nonzero\",\"globalCompositeOperation\":\"source-over\",\"transformMatrix\":null,\"skewX\":0,\"skewY\":0,\"paths\":[{\"type\":\"path\",\"originX\":\"left\",\"originY\":\"top\",\"left\":6,\"top\":6,\"width\":36,\"height\":36,\"fill\":\"#2196F3\",\"stroke\":null,\"strokeWidth\":1,\"strokeDashArray\":null,\"strokeLineCap\":\"butt\",\"strokeLineJoin\":\"miter\",\"strokeMiterLimit\":10,\"scaleX\":1,\"scaleY\":1,\"angle\":0,\"flipX\":false,\"flipY\":false,\"opacity\":1,\"shadow\":null,\"visible\":true,\"clipTo\":null,\"backgroundColor\":\"\",\"fillRule\":\"nonzero\",\"globalCompositeOperation\":\"source-over\",\"transformMatrix\":null,\"skewX\":0,\"skewY\":0,\"pathOffset\":{\"x\":24,\"y\":24},\"path\":[[\"M\",42,37],[\"C\",42,39.800781,39.800781,42,37,42],[\"L\",11,42],[\"C\",8.199219,42,6,39.800781,6,37],[\"L\",6,11],[\"C\",6,8.199219,8.199219,6,11,6],[\"L\",37,6],[\"C\",39.800781,6,42,8.199219,42,11],[\"Z\"]]},{\"type\":\"path\",\"originX\":\"left\",\"originY\":\"top\",\"left\":21.5,\"top\":14,\"width\":5,\"height\":19,\"fill\":\"#FFFFFF\",\"stroke\":null,\"strokeWidth\":1,\"strokeDashArray\":null,\"strokeLineCap\":\"butt\",\"strokeLineJoin\":\"miter\",\"strokeMiterLimit\":10,\"scaleX\":1,\"scaleY\":1,\"angle\":0,\"flipX\":false,\"flipY\":false,\"opacity\":1,\"shadow\":null,\"visible\":true,\"clipTo\":null,\"backgroundColor\":\"\",\"fillRule\":\"nonzero\",\"globalCompositeOperation\":\"source-over\",\"transformMatrix\":null,\"skewX\":0,\"skewY\":0,\"pathOffset\":{\"x\":24,\"y\":23.5},\"path\":[[\"M\",26,33],[\"L\",22,33],[\"L\",22,22],[\"L\",26,22],[\"Z\"],[\"M\",26.5,16.5],[\"C\",26.5,15.101563,25.398438,14,24,14],[\"C\",22.601563,14,21.5,15.101563,21.5,16.5],[\"C\",21.5,17.898438,22.601563,19,24,19],[\"C\",25.398438,19,26.5,17.898438,26.5,16.5]]}]}]}","type":"path-group","url":"data:image/svg+xml;utf8,<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" viewBox=\"0 0 48 48\" version=\"1.1\">\n<g id=\"surface1\">\n<path style=\" fill:#2196F3;\" d=\"M 42 37 C 42 39.800781 39.800781 42 37 42 L 11 42 C 8.199219 42 6 39.800781 6 37 L 6 11 C 6 8.199219 8.199219 6 11 6 L 37 6 C 39.800781 6 42 8.199219 42 11 Z \"/>\n<path style=\" fill:#FFFFFF;\" d=\"M 26 33 L 22 33 L 22 22 L 26 22 Z M 26.5 16.5 C 26.5 15.101563 25.398438 14 24 14 C 22.601563 14 21.5 15.101563 21.5 16.5 C 21.5 17.898438 22.601563 19 24 19 C 25.398438 19 26.5 17.898438 26.5 16.5 \"/>\n</g>\n</svg>\n"},"objectCaching":true,"opacity":1,"originX":"center","originY":"center","paths":[{"angle":0,"backgroundColor":"","clipTo":null,"editable":true,"evented":true,"fill":"#2196F3","fillRule":"nonzero","flipX":false,"flipY":false,"globalCompositeOperation":"source-over","height":36,"hoverCursor":"move","left":6,"lockScalingFlip":false,"lockUniScaling":false,"objectCaching":true,"opacity":1,"originX":"left","originY":"top","path":[["M",42,37],["C",42,39.800781,39.800781,42,37,42],["L",11,42],["C",8.199219,42,6,39.800781,6,37],["L",6,11],["C",6,8.199219,8.199219,6,11,6],["L",37,6],["C",39.800781,6,42,8.199219,42,11],["Z"]],"pathOffset":{"x":24,"y":24},"scaleX":1,"scaleY":1,"selectable":true,"shadow":null,"skewX":0,"skewY":0,"stroke":null,"strokeDashArray":null,"strokeLineCap":"butt","strokeLineJoin":"miter","strokeMiterLimit":10,"strokeWidth":1,"top":6,"transformMatrix":null,"type":"path","visible":true,"width":36},{"angle":0,"backgroundColor":"","clipTo":null,"editable":true,"evented":true,"fill":"#FFFFFF","fillRule":"nonzero","flipX":false,"flipY":false,"globalCompositeOperation":"source-over","height":19,"hoverCursor":"move","left":21.5,"lockScalingFlip":false,"lockUniScaling":false,"objectCaching":true,"opacity":1,"originX":"left","originY":"top","path":[["M",26,33],["L",22,33],["L",22,22],["L",26,22],["Z"],["M",26.5,16.5],["C",26.5,15.101563,25.398438,14,24,14],["C",22.601563,14,21.5,15.101563,21.5,16.5],["C",21.5,17.898438,22.601563,19,24,19],["C",25.398438,19,26.5,17.898438,26.5,16.5]],"pathOffset":{"x":24,"y":23.5},"scaleX":1,"scaleY":1,"selectable":true,"shadow":null,"skewX":0,"skewY":0,"stroke":null,"strokeDashArray":null,"strokeLineCap":"butt","strokeLineJoin":"miter","strokeMiterLimit":10,"strokeWidth":1,"top":14,"transformMatrix":null,"type":"path","visible":true,"width":5}],"scaleX":6.25,"scaleY":6.25,"selectable":true,"shadow":null,"skewX":0,"skewY":0,"stroke":null,"strokeDashArray":null,"strokeLineCap":"butt","strokeLineJoin":"miter","strokeMiterLimit":10,"strokeWidth":1,"top":540,"transformMatrix":null,"type":"path-group","visible":true,"width":48}]}`

**Comment by asturur** [comment]
ehm what i should see?

I get a blue icon with an white i inside.
2 paths in total. is that right?

**Comment by asturur** [comment]
```javascript
    fabric.PathGroup = { };
    fabric.PathGroup.fromObject = function(object, callback) {
      var originalPaths = object.paths;
      delete object.paths;
      if (typeof originalPaths === 'string') {
        fabric.loadSVGFromURL(originalPaths, function (elements) {
          var pathUrl = originalPaths;
          var group = fabric.util.groupSVGElements(elements, object, pathUrl);
          object.paths = originalPaths;
          callback(group);
        });
      }
      else {
        fabric.util.enlivenObjects(originalPaths, function(enlivenedObjects) {
          enlivenedObjects.forEach(function(obj) {
            obj._removeTransformMatrix();
          })
          var group = new fabric.Group(enlivenedObjects, object);
          object.paths = originalPaths;
          callback(group);
        });
      }
    };
```

This is a function that loads a path-group but gives you back a group.
Can you try that?

**Comment by blobinabottle** [comment]
Excellent this is working! You're a magician.

**Comment by asturur** [comment]
ok so i  close this and i put this function in the breaking changes list of things.

**Comment by edmaybert** [comment]
> ```js
>     fabric.PathGroup = { };
>     fabric.PathGroup.fromObject = function(object, callback) {
>       var originalPaths = object.paths;
>       delete object.paths;
>       if (typeof originalPaths === 'string') {
>         fabric.loadSVGFromURL(originalPaths, function (elements) {
>           var pathUrl = originalPaths;
>           var group = fabric.util.groupSVGElements(elements, object, pathUrl);
>           object.paths = originalPaths;
>           callback(group);
>         });
>       }
>       else {
>         fabric.util.enlivenObjects(originalPaths, function(enlivenedObjects) {
>           enlivenedObjects.forEach(function(obj) {
>             obj._removeTransformMatrix();
>           })
>           var group = new fabric.Group(enlivenedObjects, object);
>           object.paths = originalPaths;
>           callback(group);
>         });
>       }
>     };
> ```
> 
> This is a function that loads a path-group but gives you back a group.
> Can you try that?

Where is this code fix placed? I am using fabric 3.6.6 having upgraded from 1.7.11 and not sure where the code gets injected.

**Comment by asturur** [comment]
you run this code after importing fabric.
